api/v1/db/oracledb/player-dao.js

```diff
@@ -0,0 +1,49 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+
+const { serializePlayers } = require('../../serializers/players-serializer');
+
+const conn = appRoot.require('api/v1/db/oracledb/connection');
+
+const mergeRawPlayers = (rawPlayers) => {
+  const groupedRawPlayers = _.groupBy(rawPlayers, 'PLAYER_ID');
+  const mergedRawPlayers = _.map(groupedRawPlayers, (playerMetaDataArray) => {
+    playerMetaDataArray[0].playerCards = playerMetaDataArray[0].CARD_NUMBER === null ? []
+      : _.map(playerMetaDataArray, data => ({
+        cardNumber: data.CARD_NUMBER,
+        cardSuit: data.SUIT,
+      }));
+    return playerMetaDataArray[0];
+  });
+  return mergedRawPlayers;
+};
+
+const getPlayersByGameId = async (id, pid) => {
+  const connection = await conn.getConnection();
+  try {
+    const sqlParams = pid ? [id, pid] : [id];
+    const sqlQuery = `
+    SELECT P.PLAYER_ID, P.MEMBER_ID, M.MEMBER_NICKNAME, M.MEMBER_LEVEL, M.MEMBER_EXP_OVER_LEVEL, P.PLAYER_BET, S.STATUS AS PLAYER_STATUS, CN.CARD_NUMBER, CS.SUIT
+    FROM PLAYERS P 
+    INNER JOIN GAMES G ON G.GAME_ID = P.GAME_ID 
+    INNER JOIN STATUSES S ON S.STATUS_ID = P.STATUS_ID
+    INNER JOIN MEMBERS M ON M.MEMBER_ID = P.MEMBER_ID 
+    LEFT OUTER JOIN PLAYER_CARDS PC ON PC.PLAYER_ID = P.PLAYER_ID 
+    LEFT OUTER JOIN CARDS C ON PC.CARD_ID = C.CARD_ID 
+    LEFT OUTER JOIN CARD_NUMBERS CN ON C.CARD_NUMBER_ID = CN.CARD_NUMBER_ID 
+    LEFT OUTER JOIN CARD_SUITS CS ON C.CARD_SUIT_ID = CS.SUIT_ID  
+    WHERE G.GAME_ID = :id 
+    ${pid ? 'AND P.PLAYER_ID = :pid' : ''}
```

Comment:
I like your thinking on refactoring this but I think there should still be a separate `getPlayerByGameIdAndPlayerId` function to be more clear. You can still reuse most of this query string though. Also, when doing a `getXById` sql query, there should be some additional logic to check that the response contains exactly one row. Could you break this out into a separate method and add this validation?